python-avd/pyavd/_eos_designs/shared_utils/overlay.py

```diff
@@ -33,6 +33,9 @@ def vtep_loopback(self: SharedUtilsProtocol) -> str:
     def evpn_role(self: SharedUtilsProtocol) -> str | None:
         if self.underlay_router:
             default_evpn_role = self.node_type_key_data.default_evpn_role
+            if self.is_wan_router and not self.inputs.wan_use_evpn_node_settings_for_lan:
+                # For WAN routers without the knob, evpn_role should be ignored.
+                return None
```

Comment:
```suggestion
            if self.is_wan_router:
                default_evpn_role = "none"
                if not self.inputs.wan_use_evpn_node_settings_for_lan:
                    # For WAN routers without the knob, evpn_role should be ignored.
                    return None
```
Because you haven't changed the values in node_type_keys, need to set default to None here, o/w even though role is not set when toggle is enabled, we will receive role

Reply:
Same in overlay_routing_protocol

Reply:
We will change the default in 6.0.0 - in the meantime users can adjust the variable as needed on the node settings when they opt-in